tests/metagpt/utils/test_flatten_json_structure_json.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+import json5
+import re
+
+
+def flatten_json_object(obj, parent_key='', sep=', '):
+    if isinstance(obj, str):
+        return dict([("value", obj)])
+    
+    if isinstance(obj, list):
+        return dict([("value", sep.join(str(v) for v in obj))])
+    
+    items = []
+    for key, value in obj.items():
+        new_key = f"{parent_key}{sep}{key}" if parent_key else key
+        if isinstance(value, dict):
+            items.extend(flatten_json_object(value, new_key, sep=sep).items())
+        elif isinstance(value, list):
+            items.append((new_key, sep.join(str(v) for v in value)))
+        else:
+            items.append((new_key, value))
+    return dict(items)
+
+def flatten_json_structure(json_array):
+    if (isinstance(json_array, list) and len(json_array) == 1 and not isinstance(json_array[0], str)):
+        return flatten_json_structure(json_array[0])
```

Comment:
the condition here do you want to deal the array like `[[{"xx", ""}]]`?

but here
>>> t1={'name': 'John', 'age': "30", 'city': 'New York'}
>>> t2=[{'name': 'John', 'age': '30', 'city': 'New York'}]
>>> t3=[{'name': 'John', 'age': '30', 'city': 'New York'}, {'k': 'g'}]
>>> flatten_json_structure(t1)
['John', '30', 'New York']
>>> flatten_json_structure(t2)
['John', '30', 'New York']
>>> flatten_json_structure(t3)
['John, 30, New York', 'g']

it seems that a dict should be flatten into 'value1, value2,..' like t3, but it got a list of t1 or t2.

Reply:
Indeed, in my case, it is used to obtain the flattened value of a dictionary.

Reply:
> Indeed, in my case, it is used to obtain the flattened value of a dictionary.

yes, but t3 t2 got different result with same element-0 dict. 
And `flatten_json_object` function implement and so on are repeated in `test_flatten* test_try_par*` files, maybe you should remove them and use `from .. import ` instead.

Reply:
Due to my prompt design, I added the original code in the test code to avoid hallucination.

Reply:
It's fine to add it in your auto-generate unittest stage, but there is no need to commit it.